### PR TITLE
Change package name, and move PPX executable

### PR DESCRIPTION
**Pull request by mrmurphy**
Three things in this commit! I initially tried to split
it into couple of commits, but changes ended up
overlapping so I threw them all into one PR.

- Change the package name to `decco`
- Move the ppx executable to the root level
  and called it `ppx`.  (see reasoning below)
- Change the ppx executable to use bash
  instead of Node for perffffffffffffffff.

Notes on the moving of the PPX:

The current state of bucklescript ppxs is rather confusing,
each having its own executable path. Every time the user
wants to install a PPX, they have to check the readme to see
what particular location they have to point to for the
executable. With `bs-let` I moved the executable to the
root level and named it `ppx`, so that when the user is
installing, all they have to do is add `package_name/ppx` to
the ppx flags to get it installed and working. If all bucklescript
PPXs followed this pattern, users wouldn't need to refer to the
readme at all after the first few times.

What do you think? @ryb73

### CODE REVIEW

File: ppx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Get the directory of the script to work from.
```
Comment (ryb73) [inline ppx]: This should have a shebang: `#!/usr/bin/env bash`.
I actually just had a problem with this when trying to use `bs-let` on Travis; it gives me this error:
```
/home/travis/build/ryb73/prom-ex/node_modules/bs-let/ppx: 7: [: Linux: unexpected operator
/home/travis/build/ryb73/prom-ex/node_modules/bs-let/ppx: 10: [: Linux: unexpected operator
```

File: README.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
-# ppx_decco
+# Decco
 
 ## Latest version: v0.2
 **decco has been updated to support BS6!** If you need to use decco with BS5, install version ^0.1.0.
 
 ## What is it?
 
-Bucklescript PPX which generates JSON serializers and deserializers for user-defined types.
+A ReasonML / OCaml PPX which generates JSON serializers and deserializers for user-defined types.
```
Comment (ryb73) [inline README.md]: I think it's good to say BuckleScript instead of just Reason/OCaml since decco only supports BuckleScript and not native.

**Comment by ryb73** [comment]
Looks good! Just a couple comments.

File: README.md (hunk 2 above)
Comment (mrmurphy) [reply README.md]: Oh duh, you're totally right. I completely forgot that it uses Js.Json under the hood.

File: ppx (hunk 1 above)
Comment (mrmurphy) [reply ppx]: Oh, interesting. I haven't tried deploying with this yet, only using it locally. What shell is running on Travis? Bash?

File: package.json [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
+    "ppx": "./ppx"
   },
   "files": [
     "/bsconfig.json",
     "/src",
-    "/ppx/ppx_decco.sh",
-    "/ppx/ppx-linux.exe",
-    "/ppx/ppx-osx.exe"
+    "/ppx",
+    "/ppx_decco.sh",
```
[the remark this answers is not in the document]
Comment (mrmurphy) [reply package.json]: You're right. Thanks!

File: ppx (hunk 1 above)
Comment (mrmurphy) [reply ppx]: I found on this repo:
https://github.com/chartmuseum/helm-push/issues/7

that it looks like using a double equals causes issues on some shells. I'll make the change on bs-let as well.

**Comment by mrmurphy** [comment]
@ryb73 thanks for the review! I addressed your comments. Want to take another look?

File: package.json [hunk 4]
```diff
@@ -1,32 +1,29 @@
 {
-  "name": "@ryb73/decco",
-  "version": "0.2.2",
+  "name": "decco",
+  "version": "0.2.3",
```
[the remark this answers is not in the document]
Comment (mrmurphy) [reply package.json]: Good idea.

**Pull request merged**